src/algebra/monoid_algebra/basic.lean

```diff
@@ -1094,6 +1103,15 @@ lemma of'_eq_of [add_zero_class G] (a : G) : of' k G a = of k G a := rfl
 lemma of_injective [nontrivial k] [add_zero_class G] : function.injective (of k G) :=
 λ a b h, by simpa using (single_eq_single_iff _ _ _ _).mp h
 
+/--
+Embedding of the product of a `semiring k` and a `add_zero_class G` into the
```

Comment:
"embedding" is rather misleading: This is not injective, as `single_hom (0, g)` and `single_hom (0, h)` are equal for all `g` and `h`.